src/DistributedMetadata/DDLService.cpp

```diff
@@ -260,6 +266,31 @@ Int32 DDLService::doDDL(
     return err;
 }
 
+void DDLService::doDDLOnHosts(const std::vector<Poco::URI> & target_hosts, const String & payload,
+                              const String & method, const String & query_id, const String & user) const
+{
+    std::vector<String> failed_hosts;
+    /// FIXME : Parallelize doDDL on the uris
+    for (auto uri : target_hosts)
+    {
+        uri.addQueryParameter("distributed_ddl", "false");
+        const auto & err = doDDL(payload, uri, method, query_id, user);
+        if (err != ErrorCodes::OK)
+        {
+            failed_hosts.push_back(uri.getHost() + ":" + toString(uri.getPort()));
```

Comment:
`uri.toString()` ?

Reply:
do we need break the loop if we encounter err or we continue to execute the rest request to other hosts ?

Reply:
1. uri.toString() will print the full url.
2. Currently, we will continue to execute the rest request to other hosts. We can modify the logic when we import rollback feature here.

Reply:
What's the point of continuing the execution for now ? Can we just fail the operation now ?

Reply:
We can fix this behavior in a different PR